x-pack/plugins/security/public/views/management/edit_role/components/privileges/kibana/privilege_space_table.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiInMemoryTable,
+  EuiButtonIcon,
+} from '@elastic/eui';
+import { PrivilegeSelector } from './privilege_selector';
+import { SpaceAvatar } from '../../../../../../../../spaces/public/views/components/space_avatar';
+import { isReservedRole } from '../../../../../../lib/role';
+
+export class PrivilegeSpaceTable extends Component {
+  state = {
+    searchTerm: ''
+  }
+
+  render() {
+    const {
+      role,
+      spaces,
+      availablePrivileges,
+      spacePrivileges,
+    } = this.props;
+
+    const {
+      searchTerm
+    } = this.state;
+
+    const allTableItems = Object.keys(spacePrivileges).map(spaceId => {
+      return {
+        space: spaces.find(s => s.id === spaceId),
+        privilege: spacePrivileges[spaceId][0]
+      };
+    }).filter(item => !!item.space);
+
+    const visibleTableItems = allTableItems.filter(item => item.space.name.toLowerCase().indexOf(searchTerm) >= 0);
+
+    if (allTableItems.length === 0) {
+      return null;
+    }
+
+    return (
+      <EuiInMemoryTable
+        hasActions
+        columns={this.getTableColumns(role, availablePrivileges)}
+        search={{
+          box: {
+            incremental: true,
+            placeholder: 'Filter...',
+          },
+          onChange: (search) => {
+            this.setState({
+              searchTerm: search.queryText.toLowerCase()
+            });
+          }
+        }}
+        items={visibleTableItems}
+      />
+    );
+  }
+
+  getTableColumns = (role, availablePrivileges) => {
+    const columns = [{
+      field: 'space',
+      name: 'Space',
+      width: '50%',
+      render: (space) => (
+        <EuiFlexGroup gutterSize="s" responsive={false} alignItems={'center'}>
+          <EuiFlexItem grow={false}><SpaceAvatar space={space} size="s" /></EuiFlexItem>
+          <EuiFlexItem><EuiText>{space.name}</EuiText></EuiFlexItem>
+        </EuiFlexGroup>
+      )
+    }, {
+      field: 'privilege',
+      name: 'Privilege',
+      render: (privilege, record) => {
+        if (this.props.readonly) {
```

Comment:
Can you reduce the width of this column if it is read only? That will give more space to the space name.